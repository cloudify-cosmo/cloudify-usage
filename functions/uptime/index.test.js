const { getGeoLocationInfo } = require('./index');

describe('getGeoLocationInfo', () => {
    // NOTE: some random IP addresses
    const ipsToResolveCorrectly = [
        '212.180.218.141',
        '3.45.34.252',
        '90.251.171.80',
        '212.82.41.93',
        '157.69.217.108',
    ];

    ipsToResolveCorrectly.forEach((ip) => {
        it(`should resolve IP information for ${ip}`, async () => {
            expect(await getGeoLocationInfo(ip)).toMatchSnapshot();
        });
    });

    it('should report an unknown problem for some problematic IP', () => {
        return expect(
            getGeoLocationInfo('248.47.119.137')
        ).rejects.toThrowError();
    });

    it('should report an empty result when the IP cannot be matched', async () => {
        expect(await getGeoLocationInfo('248.')).toStrictEqual({});
    });
});
