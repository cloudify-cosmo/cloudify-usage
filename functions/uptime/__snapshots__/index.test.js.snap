// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGeoLocationInfo should resolve IP information for 3.45.34.252 1`] = `
Object {
  "city": "Seattle",
  "continent": "North America",
  "country": "United States",
  "location": "Seattle, Washington, United States",
  "subdivision": "Washington",
  "timezone": "America/Los_Angeles",
}
`;

exports[`getGeoLocationInfo should resolve IP information for 90.251.171.80 1`] = `
Object {
  "city": "",
  "continent": "",
  "country": "United Kingdom",
  "location": ", United Kingdom",
  "subdivision": "",
  "timezone": "Europe/London",
}
`;

exports[`getGeoLocationInfo should resolve IP information for 157.69.217.108 1`] = `
Object {
  "city": "",
  "continent": "",
  "country": "Japan",
  "location": ", Japan",
  "subdivision": "",
  "timezone": "Asia/Tokyo",
}
`;

exports[`getGeoLocationInfo should resolve IP information for 212.82.41.93 1`] = `
Object {
  "city": "",
  "continent": "",
  "country": "Germany",
  "location": ", Germany",
  "subdivision": "",
  "timezone": undefined,
}
`;

exports[`getGeoLocationInfo should resolve IP information for 212.180.218.141 1`] = `
Object {
  "city": "Warsaw",
  "continent": "Europe",
  "country": "Poland",
  "location": "Warsaw, Mazovia, Poland",
  "subdivision": "Mazovia",
  "timezone": "Europe/Warsaw",
}
`;
